esp/logging/loggingagent/espserverloggingagent/loggingagent.cpp

```diff
@@ -53,6 +53,33 @@ bool CESPServerLoggingAgent::init(const char * name, const char * type, IPropert
     maxServerWaitingSeconds = cfg->getPropInt(PropServerWaitingSeconds);
     maxGTSRetries = cfg->getPropInt(MaxTriesGTS, DefaultMaxTriesGTS);
 
+    Owned<IPropertyTreeIterator> iter = cfg->getElements("LogSourceMap/LogSource");
+    ForEach(*iter)
+    {
+        StringBuffer name, groupName, dbName, transactionSeed, statusMessage;
+        ensureInputString(iter->query().queryProp("@name"), false, name, -1, "LogSource @name required");
+        ensureInputString(iter->query().queryProp("@maptologgroup"), true, groupName, -1, "LogSource @maptologgroup required");
+        ensureInputString(iter->query().queryProp("@maptodb"), true, dbName, -1, "LogSource @maptodb required");
+        Owned<CLogSource> logSource = new CLogSource(name.str(), groupName.str(), dbName.str());
+        logSources.setValue(name.str(), logSource);
+
+        CTransID* transID = transIDMap.getValue(groupName.str());
+        if (!transID)
+        {
+            getTransactionSeed(groupName.str(), transactionSeed, statusMessage);
+            if (transactionSeed.length() == 0)
+            {
+                VStringBuffer msg("Unable to get transaction seed for LogSource:%s", groupName.str());
+                if (statusMessage.length())
+                    msg.append(" - ").append(statusMessage.str());
```

Comment:
I'm not sure if it matters much, but rather than 2 append calls, it might be advantageous to perform a formatted append.... appendf(" - %s", statusMessage.str())